src/berry_blender.c

```diff
@@ -0,0 +1,3883 @@
+	strh r1, [r7, 0x6]\n\
+	subs r0, r3\n\
+	strh r0, [r7, 0x8]\n\
+	movs r3, 0\n\
+	movs r2, 0\n\
+	adds r1, r7, 0\n\
+	movs r6, 0x4\n\
+_080505AA:\n\
+	movs r4, 0\n\
+	ldrsh r0, [r1, r4]\n\
+	cmp r0, 0\n\
+	bge _080505B6\n\
+	strh r2, [r1]\n\
+	adds r3, 0x1\n\
+_080505B6:\n\
+	adds r1, 0x2\n\
+	subs r6, 0x1\n\
+	cmp r6, 0\n\
+	bge _080505AA\n\
+	lsls r0, r3, 24\n\
+	lsrs r0, 24\n\
+	mov r10, r0\n\
+	movs r4, 0\n\
+	ldr r1, _080505DC @ =gUnknown_03000510\n\
+	movs r6, 0x4\n\
+_080505CA:\n\
+	ldrh r2, [r1]\n\
+	movs r5, 0\n\
+	ldrsh r0, [r1, r5]\n\
+	cmp r0, 0\n\
+	ble _080505E4\n\
+	cmp r0, r3\n\
+	bge _080505E0\n\
+	strh r4, [r1]\n\
+	b _080505E4\n\
+	.align 2, 0\n\
+_080505DC: .4byte gUnknown_03000510\n\
+_080505E0:\n\
+	subs r0, r2, r3\n\
+	strh r0, [r1]\n\
+_080505E4:\n\
+	adds r1, 0x2\n\
+	subs r6, 0x1\n\
+	cmp r6, 0\n\
+	bge _080505CA\n\
+	ldr r1, _080506C4 @ =gUnknown_03000510\n\
+	ldr r2, _080506C8 @ =gUnknown_03000530\n\
+	movs r6, 0x4\n\
+_080505F2:\n\
+	movs r3, 0\n\
+	ldrsh r0, [r1, r3]\n\
+	stm r2!, {r0}\n\
+	adds r1, 0x2\n\
+	subs r6, 0x1\n\
+	cmp r6, 0\n\
+	bge _080505F2\n\
+	ldr r1, _080506CC @ =0x0000014d\n\
+	ldr r0, [sp, 0x8]\n\
+	bl __udivsi3\n\
+	lsls r0, 16\n\
+	lsrs r0, 16\n\
+	adds r3, r0, 0\n\
+	adds r3, 0x64\n\
+	ldr r4, _080506D0 @ =gUnknown_0300055C\n\
+	str r3, [r4]\n\
+	movs r6, 0x4\n\
+_08050616:\n\
+	movs r0, 0\n\
+	ldrsh r5, [r7, r0]\n\
+	adds r0, r5, 0\n\
+	muls r0, r3\n\
+	movs r1, 0xA\n\
+	str r3, [sp, 0xC]\n\
+	bl __divsi3\n\
+	adds r5, r0, 0\n\
+	movs r1, 0xA\n\
+	bl __modsi3\n\
+	adds r4, r0, 0\n\
+	adds r0, r5, 0\n\
+	movs r1, 0xA\n\
+	bl __divsi3\n\
+	adds r5, r0, 0\n\
+	ldr r3, [sp, 0xC]\n\
+	cmp r4, 0x4\n\
+	ble _08050642\n\
+	adds r5, 0x1\n\
+_08050642:\n\
+	strh r5, [r7]\n\
+	adds r7, 0x2\n\
+	subs r6, 0x1\n\
+	cmp r6, 0\n\
+	bge _08050616\n\
+	ldr r1, _080506C4 @ =gUnknown_03000510\n\
+	ldr r2, _080506D4 @ =gUnknown_03000548\n\
+	movs r6, 0x4\n\
+_08050652:\n\
+	movs r3, 0\n\
+	ldrsh r0, [r1, r3]\n\
+	stm r2!, {r0}\n\
+	adds r1, 0x2\n\
+	subs r6, 0x1\n\
+	cmp r6, 0\n\
+	bge _08050652\n\
+	ldr r4, _080506C4 @ =gUnknown_03000510\n\
+	ldr r0, [sp]\n\
+	adds r1, r4, 0\n\
+	mov r2, r9\n\
+	mov r3, r10\n\
+	bl Blender_GetPokeblockColor\n\
+	mov r5, r8\n\
+	strb r0, [r5]\n\
+	movs r1, 0xA\n\
+	ldrsh r0, [r4, r1]\n\
+	mov r1, r9\n\
+	bl __divsi3\n\
+	mov r3, r9\n\
+	subs r0, r3\n\
+	strh r0, [r4, 0xA]\n\
+	lsls r0, 16\n\
+	cmp r0, 0\n\
+	bge _0805068C\n\
+	movs r0, 0\n\
+	strh r0, [r4, 0xA]\n\
+_0805068C:\n\
+	mov r5, r8\n\
+	ldrb r0, [r5]\n\
+	cmp r0, 0xC\n\
+	bne _080506E6\n\
+	bl Random\n\
+	lsls r0, 16\n\
+	lsrs r0, 16\n\
+	movs r1, 0xA\n\
+	bl __umodsi3\n\
+	lsls r0, 16\n\
+	lsrs r3, r0, 16\n\
+	movs r6, 0\n\
+	ldr r0, _080506D8 @ =gUnknown_082165DF\n\
+	adds r0, r3, r0\n\
+	ldrb r0, [r0]\n\
+	adds r1, r4, 0\n\
+	movs r4, 0x1\n\
+	movs r3, 0x2\n\
+_080506B4:\n\
+	adds r2, r0, 0\n\
+	asrs r2, r6\n\
+	ands r2, r4\n\
+	cmp r2, 0\n\
+	beq _080506DC\n\
+	strh r3, [r1]\n\
+	b _080506DE\n\
+	.ali
```

Comment:
Yeah, I tried too and couldn't get the `frames` stuff to match. Not sure what's happening with `max_RPM`, either because the `if (max_RPM <= 64)` comparison basically invalidates most of that block. I guess this is going to be a fix later thing.